### PR TITLE
Ray: Add distanceSqToPoint method

### LINKED ISSUE #3385: Add Ray.distanceToPointSquared method
I suggest add the squared version of Ray.distanceToPoint, just like Vector3.distanceTo to Vector3.distanceToSquared, to prevent unwanted computation in some situation.

Not a big problem,I can add it in my project myself though. : )

### PR DESCRIPTION
**Pull request by dubejf**
Fixes #3385. Replaces PR #6757.

### CODE REVIEW

File: src/objects/Sprite.js [hunk 1]
```diff
@@ -38,9 +38,9 @@ THREE.Sprite.prototype.raycast = ( function () {
 
 		matrixPosition.setFromMatrixPosition( this.matrixWorld );
 
-		var distance = raycaster.ray.distanceToPoint( matrixPosition );
+		var distanceSq = raycaster.ray.distanceSqToPoint( matrixPosition );
 
-		if ( distance > this.scale.x ) {
+		if ( distanceSq > this.scale.x * this.scale.x ) {
```
Comment (WestLangley) [inline src/objects/Sprite.js]: This line in the sprite raycasting code,

    if ( distance > this.scale.x ) {

is a hack. It does not take into consideration the shape of the sprite, or the sprite's transparent areas.

Given that it is a hack, and given that we are now using the squared distance, perhaps we should do this instead:

    if ( distanceSq > this.scale.x * this.scale.y ) {

Comment (dubejf) [inline src/objects/Sprite.js]: What about
```
var diag = ( this.scale.x + this.scale.y ) / 2;
if ( distanceSq > diag * diag ) 
```

Comment (WestLangley) [inline src/objects/Sprite.js]: It's a hack. It doesn't matter. That's not a diagonal, though.

File: src/math/Ray.js [hunk 2]
```diff
@@ -71,6 +71,16 @@ THREE.Ray.prototype = {
 
 	distanceToPoint: function () {
 
+		return function ( point ) {
```
Comment (mrdoob) [inline src/math/Ray.js]: Shouldn't this just be this?

```javascript
	distanceToPoint: function ( point ) {

		return Math.sqrt( this.distanceSqToPoint( point ) );

	},
```

Comment (dubejf) [inline src/math/Ray.js]: Of course!

**Comment by mrdoob** [comment]
Thanks thanks!

**Pull request merged**